### PR TITLE
Added support for floating point write

### PR DESCRIPTION
**Pull request by zyvitski**
ISSUE #10

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -562,3 +562,46 @@ fn write_read_i24_is_lossless() {
...
+        for (expected, read) in (1..257)
+            .map(|x| 1.0_f32 / x as f32)
+            .zip(reader.samples()) {
+            assert_eq!(expected, read.unwrap());
+        }
+    }
+}
+#[test]
+#[should_panic]
+fn test_name() {
```
Comment (ruuda) [inline src/lib.rs]: :+1: for also testing the error path. The name should be more descriptive though, something like `no_32_bps_for_float_sample_format_panics`?

File: src/write.rs [hunk 2]
```diff
@@ -216,9 +216,8 @@ impl<W> WavWriter<W>
             try!(buffer.write_le_u16(spec.channels));
             // The field nSamplesPerSec.
             try!(buffer.write_le_u32(spec.sample_rate));
-            let bytes_per_sec = spec.sample_rate
-                              * self.bytes_per_sample as u32
-                              * spec.channels as u32;
+            let bytes_per_sec = spec.sample_rate * self.bytes_per_sample as u32 *
+                                spec.channels as u32;
```
Comment (ruuda) [inline src/write.rs]: Please avoid formatting-only changes, they make `git blame` less useful. (Also below.)

You can use `git reset --patch master` to interactively undo these formatting-only changes, and then a `git commit --amend` to update the commit such that these changes never happened.

File: src/write.rs [hunk 3]
```diff
@@ -234,8 +233,21 @@ impl<W> WavWriter<W>
...
+            match spec.sample_format {
+                //We use PCM audio with integer samples.
+                SampleFormat::Int => try!(buffer.write_all(&super::KSDATAFORMAT_SUBTYPE_PCM)),
+
+                SampleFormat::Float => {
+                    if spec.bits_per_sample == 32 {
+                        try!(buffer.write_all(&super::KSDATAFORMAT_SUBTYPE_IEEE_FLOAT));
+                    } else {
+                        //TODO: better error condition
+                        panic!("Invalid number of bits for floating point sample.");
```
Comment (ruuda) [inline src/write.rs]: There is still a to do in here. We should change the message to something that also indicates how to fix the problem, something along the lines of `Invalid number of bits per sample. When writing SampleFormat::Float, bits_per_sample must be 32.`.

File: src/write.rs [hunk 4]
```diff
@@ -259,9 +271,6 @@ impl<W> WavWriter<W>
     #[inline]
     pub fn write_sample<S: Sample>(&mut self, sample: S) -> Result<()> {
         // For now, only the integer PCM format is supported for writing.
```
Comment (ruuda) [inline src/write.rs]: This comment is can now be removed.

File: src/write.rs (hunk 2 above)
Comment (zyvitski) [inline src/write.rs]: Sorry about that, I have my environment setup to auto format using rustfmt on save. I will correct it.

**Comment by zyvitski** [comment]
I will go through and make all of these changes either tonight or tomorrow afternoon.

File: src/lib.rs (hunk 1 above)
Comment (zyvitski) [inline src/lib.rs]: Looks like I forgot to change the name after my ide inserted the test snippet. Will rename.

**Comment by zyvitski** [comment]
Ok, I went through and fixed up the items we discussed.

**Comment by ruuda** [comment]
Thanks a lot for doing this!

I cherry-picked the last commit as d20bfdd.

**Pull request closed without merging**